src/Hocon.Configuration.Test/ConcurrentImmutablitySpec.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Xunit;
+
+namespace Hocon.Configuration.Tests
+{
+    public class ConcurrentImmutablitySpec
+    {
+
+        private static Config fb2 = HoconConfigurationFactory.ParseString(@"akka.actor{
+                dispatchers{
+                    foo.test = bar
+                    foo2.test = baz
+                }
+            }");
+
+        private static Config fb1 = HoconConfigurationFactory.ParseString(@"akka.actor{
+                deployment{
+                    /foo{
+                        dispatcher = akka.actor.dispatchers.foo
+                    }
+                }
+                dispatchers{
+                    foo2.test = biz
+                }
+            }");
+
+        private static Config fb3 = HoconConfigurationFactory.ParseString(@"akka.actor{
+            deployment{
+                /harmless{
+                    throughput = 12
+                }
+            }
+            dispatchers{
+                foo3.test = boz
+            }
+        }");
+
+        private Config C { get; set; } = fb1;
+
+        [Fact]
+        public async Task ConcurrentConfig_modification_should_be_immutable()
+        {
+            C = C.SafeWithFallback(fb2);
+
+            var r = Parallel.ForEach(Enumerable.Range(1, 100), i =>
+            {
+                /*
+                 * Modify the Config that is being concurrently read with a fallback
+                 * that doesn't modify any of the properties actually being read.
+                 *
+                 * This operation should not throw - every other thread should be reading its
+                 * own immutable copy of the object.
+                 */
+                if (i % 2 == 0)
```

Comment:
Uses this to trigger the same types of exceptions we had before. Should only throw if the underlying behavior is mutable.